src/app/Services/UserPostService.php

```diff
@@ -35,6 +36,55 @@ public function create(CreateUserPostRequest $request): void
         }         
     }
     
+    /**
+    * Edit a new user post
+    */
+    public function edit(EditUserPostRequest $request): void
+    {
+        $postToEdit = UserPost::findOrFail($request->userPostToEditId);
+        $postToEdit->update(['content' => $request->editedContent]);
+
+        if($request->shouldRemoveImage){
+            PostMedia::find($request->postMediaToEditId)->delete();
+        }
+
+        if ($request->hasFile('editedImage')) {
+            $file = $request->file('editedImage');
+            $extension = $file->getClientOriginalExtension(); 
+            $filename = time() . '.' . $extension;
+            $path = 'uploads/images/';
+            $file->move($path, $filename);
+
+            PostMedia::updateOrCreate(
+                ['post_id' => $postToEdit->id],
+                ['image' => $path . $filename]
+            );
+        }
+    }
```

Comment:
I think this can be optimized by minimizing the use of variables thus saving space

Reply:
done